File: leetcode/2288-apply-discount-to-prices/discountPrices.go
package leetcode

import (
	"fmt"
	"regexp"
	"strconv"
	"strings"
)

func discountPrices(sentence string, discount int) string {
	var words = strings.Split(sentence, " ")
	var compRegEx = regexp.MustCompile(`^\$(\d+)$`)

	for index, word := range words {
		match := compRegEx.FindStringSubmatch(word)

		if len(match) > 0 {
			price, _ := strconv.ParseFloat(match[1], 64)

			words[index] = fmt.Sprintf("$%.2f", price*(1-float64(discount)/100))
		}

	}

	return strings.Join(words, " ")
}

File: leetcode/2288-apply-discount-to-prices/discountPrices.ts
function discountPrices(sentence: string, discount: number): string {
  const words = sentence.split(" ");

  for (const [index, word] of words.entries()) {
    const match = word.match(/^\$(\d+)$/);

    if (match != null) {
      const price = (Number(match[1]) / 100) * (100 - discount);
      words[index] = "$" + price.toFixed(2);
    }
  }
  return words.join(" ");
}

export default discountPrices;
